docker/Dockerfile

```diff
@@ -9,11 +9,14 @@ COPY . /polkadot
 
 RUN apt-get update && \
 	apt-get upgrade -y && \
-	apt-get install -y cmake pkg-config libssl-dev git
-
+	apt-get install -y cmake pkg-config libssl-dev git clang
 RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
-	export PATH=$PATH:$HOME/.cargo/bin && \
-	cargo build --$PROFILE
+        export PATH=$PATH:$HOME/.cargo/bin && \
+        rustup toolchain install nightly && \
+        rustup target add wasm32-unknown-unknown --toolchain nightly && \
+        cargo install --git https://github.com/alexcrichton/wasm-gc && \
```

Comment:
Can't all of these commands to setup the rust environment be replaced by a call to `./scripts/init.sh`?

Reply:
Trying. Will update you shortly

Reply:
@agutsal you can reference (or even mention it in `FROM`) this [one](https://github.com/paritytech/scripts/blob/master/dockerfiles/rust-builder/Dockerfile). We are updating it nightly.

Reply:
@andresilva that `init.sh` is not used by us anywhere btw.

Reply:
@andresilva @TriplEight I guess you guys should come to some agreement and not waste my time. Let me know by tagging me in that thread. I'll switch notifications off in the meantime. Thank you.

Reply:
@agutsal I can't push to your branch otherwise I'd fix it myself. This works https://github.com/paritytech/polkadot/commit/060191baf6e4ff6ccb428078ecfd5e51ddc09e4c and I would merge it. I think you can allow this by clicking the `allow edits from maintainers checkbox` somewhere in this page.

@TriplEight it's not used by us but it's mentioned in the documentation of both substrate and polkadot. It's a small script that does pretty much the same thing that's being done in these lines of the Dockerfile. The intention of this PR is to fix the existing Dockerfile so let's keep the changes to that end.

Reply:
@andresilva , ok then. I will check if it will work, a bit later.

Reply:
@andresilva it was checked by default:
![image](https://user-images.githubusercontent.com/1643117/61125466-92784400-a4b2-11e9-902e-5c4fba478f9c.png)